### PR TITLE
Feature/70 story handler

### LINKED ISSUE #70: Implement story handler
## Description

Each unity scene can consist of multiple inky dialogue scripts. To make sure this is possible, the story controller should be implemented. It will hold a list of inky dialogue scripts to be played sequentially, with transitions between them (like fading to black or showing a zoomed out view, or no transition at all).

To make sure cross examination parts work properly, each of these scripts should also be flagged with a  DialogueControllerMode value, so the DialogueController can be set into the correct mode when loading the new inky script.

This script should also handle transitioning to the next unity scene when done.

## Dev Guide

- Make sure to use _onDialogueFinished event to continue when the current inky dialogue script is done
- SetNarrativeScript currently only takes a new inky dialogue script, so this should be extended to also take a DialogueControllerMode
- The first action in a script should be automatically called, which it currently isn't.

## Acceptance Criteria

This story will be finished when the aforementioned functionality is implemented and a first pre-trial unity scene is set up to test it all and set up some of the actual working behaviour going forward.

### PR DESCRIPTION
**Pull request by PhoebeMitchell**
## Summary

Implements story controller. Identical to [this PR](https://github.com/Studio-Lovelies/GG-JointJustice-Unity/pull/82) except with some added polish.

"Stories are now handled by the story handler, which can transition between different inky dialogue scripts. It also loads the next unity scene when reaching the end"

## Testing instructions

Run the ink test scene and make sure the narratives scripts transition into each other. Also run the tests.
Note: the screenshake doesn't stop when continuing to the next narrative script. This could be considered a bug (and it will need to be fixed in another PR if so), though I suspect screenshakes wont be that long usually, so it might not matter.

## Additional information

- `[ ]` Changes UI
- `[x]` Introduces new feature
- `[ ]` Removes existing feature
- `[x]` Has associated resource: 
  
closes #70

### CODE REVIEW

**Review by lyndsiWilliams — APPROVED** [review]
Code looks great and the test scene runs great! All the tests passed, nice work!

File: Assets/Scripts/TextDecoder/StoryController.cs [hunk 1]
```diff
@@ -9,11 +9,10 @@ public class StoryController : MonoBehaviour
     [SerializeField] private List<Dialogue> _dialogueList;
 
     [Header("Events")]
-
+
```
Comment (ViMaSter) [inline Assets/Scripts/TextDecoder/StoryController.cs:12]: I think this line break is superfluous and should be safe to remove.

File: Assets/Tests/PlayModeTests/Scripts/EvidenceMenu/EvidenceMenuKeyboardTests.cs [hunk 2]
```diff
@@ -21,9 +21,10 @@ public class EvidenceMenuKeyboardTests
         [ReloadScene("Assets/Scenes/EvidenceMenu - Test Scene.unity")]
         public IEnumerator EvidenceMenuOpensAndCloses()
         {
-
             yield return null;
-            global::EvidenceMenu evidenceMenu = InputTestTools.FindInactiveInScene<global::EvidenceMenu>()[0];
+            var dialogueController = Object.FindObjectOfType<global::DialogueController>();
+            yield return TestTools.WaitForState(() => !dialogueController.IsBusy);
```
Comment (ViMaSter) [inline Assets/Tests/PlayModeTests/Scripts/EvidenceMenu/EvidenceMenuKeyboardTests.cs:26]: 👀 👌

**Review by ViMaSter — APPROVED** [review]
Looks good to me! <sup>(Wouldn't be a review without a nit though. Just a minor formatting thing, feel free to skip.)</sup>

File: Assets/Scripts/TextDecoder/StoryController.cs (hunk 1 above)
Comment (PhoebeMitchell) [reply Assets/Scripts/TextDecoder/StoryController.cs:12]: Agreed that you will handle this in your branch.

**Pull request merged**